appium/webdriver/webdriver.py

```diff
@@ -581,8 +582,23 @@ def lock(self, seconds):
             'seconds': seconds,
         }
         self.execute(Command.LOCK, data)
+        if self.desired_capabilities.get('platformName', '').lower() == 'android':
+            WebDriverWait(self, 10).until(lambda d: d.is_locked())
```

Comment:
This is really up for discussion.  I personally believe that the Appium server should be responsible for this, not the client.  If the maintainers decide not to add this, that's totally understandable.  It was just helpful for "robustifying" while testing the change.
I'm only running this check on Android because iOS should be back in an unlocked state by the end of the `execute` call.

Reply:
We'd love to remove wait in each method and make users insert wait in their own code.
(Other clients also don't have `wait` in Appium client side.)